cmd/car/extract.go

```diff
@@ -92,6 +94,26 @@ func extractRoot(c *cli.Context, ls *ipld.LinkSystem, root cid.Cid, outputDir st
 		}
 	}
 	if err := extractDir(c, ls, ufn, outputResolvedDir, "/"); err != nil {
+		if err == ErrNotDir {
+			ufsData, err := pbnode.LookupByString("Data")
+			if err != nil {
+				return err
+			}
+			ufsBytes, err := ufsData.AsBytes()
+			if err != nil {
+				return err
+			}
+			ufsNode, err := data.DecodeUnixFSData(ufsBytes)
+			if err != nil {
+				return err
+			}
+			if ufsNode.DataType.Int() == data.Data_File || ufsNode.DataType.Int() == data.Data_Raw {
```

Comment:
nit: would be a bit shorter if you use a switch, as you deduplicate the left side